test.js

```diff
@@ -49,6 +49,76 @@ test('.set() - with object', t => {
 	t.is(t.context.config.get('baz.foo.bar'), 'baz');
 });
 
+test('.set() - with Date', t => {
+	t.context.config.set('dateobj', new Date());
+	const myDate = t.context.config.get('dateobj');
+	t.true(myDate instanceof Date);
+});
+
+test('.set() - with Date in a nested object', t => {
+	t.context.config.set('dateobj', {str: 'this is a string', date: new Date()});
+	const myDate = t.context.config.get('dateobj').date;
+	t.true(myDate instanceof Date);
+});
+
+test('.set() - with Date in a array', t => {
+	t.context.config.set('datearray', [new Date(), 'asd', 1337, new Date()]);
+	const myDateArray = t.context.config.get('datearray');
+	t.true(myDateArray[0] instanceof Date);
+	t.true(myDateArray[3] instanceof Date);
+});
+
+test('.set() - with Date in a object inside array', t => {
+	t.context.config.set('datearray', [{date: new Date(), dummy: 'dummy', num: -1337}, 'asd', 1337, new Date()]);
+	const myDateArray = t.context.config.get('datearray');
+	t.true(myDateArray[0].date instanceof Date);
+	t.true(myDateArray[3] instanceof Date);
+});
+
+test('.set() - with Date in a nested object inside array', t => {
+	t.context.config.set('datearray', [{nested: {date: new Date(), nestedDummy: 'nested dummy'}, dummy: 'dummy', num: -1337}, 'asd', 1337, new Date()]);
+	const myDateArray = t.context.config.get('datearray');
+	t.true(myDateArray[0].nested.date instanceof Date);
+	t.true(myDateArray[3] instanceof Date);
+});
+
+test('.set() - with Date in a nested array', t => {
+	t.context.config.set('datearray', [[1, 2, 3], 'asd', [new Date(), 1337, new Date()], new Date()]);
+	const myDateArray = t.context.config.get('datearray');
+	t.true(myDateArray[2][0] instanceof Date);
+	t.true(myDateArray[2][2] instanceof Date);
+	t.true(myDateArray[3] instanceof Date);
+});
+
+test('.set()  - with Date and circular object', t => {
+	const circular = {myDate: new Date(), circulation: null};
+	circular.circulation = circular;
+	t.context.config.set('circular', circular);
+	const getCircular = t.context.config.get('circular');
+	t.true(getCircular.myDate instanceof Date);
+	t.true(getCircular.circulation.myDate instanceof Date);
+});
+
+test('.set() - with Date and 2 level deep circular object', t => {
+	const circular = {myDate: new Date(), circulationl1: {l2: null}};
+	circular.circulationl1.l2 = circular;
+	t.context.config.set('circular', circular);
+	const getCircular = t.context.config.get('circular');
+	t.true(getCircular.myDate instanceof Date);
+	t.true(getCircular.circulationl1.l2.myDate instanceof Date);
+});
+
+test('.set() - with Date and multiple circular objects', t => {
+	const circular = {myDate: new Date(), c1: null, c2: null};
+	circular.c1 = circular;
+	circular.c2 = circular;
+	t.context.config.set('circular', circular);
+	const getCircular = t.context.config.get('circular');
+	t.true(getCircular.myDate instanceof Date);
+	t.true(getCircular.c1.myDate instanceof Date);
+	t.true(getCircular.c2.myDate instanceof Date);
+});
+
```

Comment:
There needs to be some tests for other types, just to ensure it works in general with custom types and other types than Date.